category-service/src/main/java/com/edmart/category/serviceImpl/CategoryServiceImpl.java

```diff
@@ -21,51 +26,84 @@ public class CategoryServiceImpl implements CategoryService {
-        try {
-            categoryRepository.save(CategoryMapper.INSTANCE.mapToCategory(categoryDTO));
+    public void createCategory(CategoryDTO categoryDTO) throws CategoryNotFoundException {
+        Category category = new Category();
+        try{
+            category.setCategoryName(categoryDTO.categoryName());
+            category.setCategoryDescription(categoryDTO.categoryDescription());
+            category.setCategoryDesignation(categoryDTO.categoryDesignation());
 
-        }catch (Exception e){
-            log.error("Error creating category with name {} caused by {}",categoryDTO.categoryName(),e.getCause());
+            categoryRepository.save(category);
+        }catch(Exception e){
+            log.error("Error Creating category with name {} caused by {}", categoryDTO.categoryName(), e.getCause());
         }
     }
 
     @Override
-    public List<CategoryDTO> getAllCategories() {
-        return categoryRepository.findAll().stream()
-                .map(CategoryMapper.INSTANCE::mapToCategoryDTO)
-                .collect(Collectors.toList());    }
+    public CategoryResponseDTO getAllCategories(int page, int size,String sortBy, String sortDir) {
+        Sort sort = sortDir.equalsIgnoreCase(Sort.Direction.DESC.name())
+                ? Sort.by(sortBy).ascending()
+                : Sort.by(sortBy).descending();
+
+        Pageable pageable = PageRequest.of(page, size, sort);
+
+        Page<Category> categoryPage = categoryRepository.findAll(pageable);
+
+        CategoryResponseDTO categoryResponseDto=new CategoryResponseDTO();
+        categoryResponseDto.setCategoryList(categoryPage.getContent());
+        categoryResponseDto.setPageInfoDTO(Pagination.getPageInfoDTO(categoryPage));
+
+        return  categoryResponseDto;
+    }
 
     @Override
-    public CategoryDTO getCategory(Long categoryId) {
-        return categoryRepository.findById(categoryId)
-                .map(CategoryMapper.INSTANCE::mapToCategoryDTO)
-                .orElseThrow(() -> new CategoryNotFoundException("Category not found"));
+    public CategoryDTO getCategory(Long categoryId) throws CategoryNotFoundException {
+        Optional<CategoryDTO> categoryDTOOptional = categoryRepository.findById(categoryId)
+                .map(categoryMapper);
+        if(categoryDTOOptional.isPresent()){
+            return categoryDTOOptional.get();
+        }else{
+            throw new CategoryNotFoundException("No Category exist with this Id");
+        }
     }
 
     @Override
-    public void updateCategory(Long categoryId, CategoryDTO categoryDTO) {
-        try {
-            Optional<Category> categoryOptional = categoryRepository.findById(categoryId);
-            if (categoryOptional.isEmpty()) {
-                throw new CategoryNotFoundException("Category not found");
+    public void updateCategory(Long categoryId, CategoryDTO categoryDTO) throws CategoryNotFoundException {
+        try{
```

Comment:
There is better way to do this now, this is by using BeanWrapper class. This way is better because you dont have to do so many if....else and also u dont have to remember every request parameter from the request body. Follow the implementation below to make your code clean and optimised :).

//create a seperate method called getNullPropertyNames with the request body parameter
private String[] getNullPropertyNames(CategoryDTO categoryDTO)  {
        BeanWrapper beanWrapper = new BeanWrapperImpl(categoryDTO);
        PropertyDescriptor[] propertyDescriptors = beanWrapper.getPropertyDescriptors();

        Set<String> nullProperties = new HashSet<>();
        for (PropertyDescriptor propertyDescriptor : propertyDescriptors) {
            String propertyName = propertyDescriptor.getName();
            if (beanWrapper.getPropertyValue(propertyName) == null) {
                nullProperties.add(propertyName);
            }
        }